spec.md

```diff
@@ -937,140 +783,208 @@ Property types are legal in the following constructs:
 
-An unsigned integer type is always equivalent to another unsigned integer type regardless of bit width, and is not equivalent to any other type.
-Similarly, a signed integer type is always equivalent to another signed integer type regardless of bit width, and is not equivalent to any other type.
+### Width Inference
 
-Clock types are equivalent to clock types, and are not equivalent to any other type.
+Normally, the three integer ground types are written with explicit bit widths.
+The bit widths are given in angel brackets, such as the `8`{.firrtl} in `UInt<8>`{.firrtl}.
+This is called the **uninferred** variant of the type.
 
-An uninferred `Reset`{.firrtl} can be connected to another `Reset`{.firrtl}, `UInt`{.firrtl} of unknown width, `UInt<1>`{.firrtl}, or `AsyncReset`{.firrtl}.
-It cannot be connected to both a `UInt`{.firrtl} and an `AsyncReset`{.firrtl}.
+FIRRTL also supports an **inferred** variant of these types.
+They are written as follows:
 
-The `AsyncReset`{.firrtl} type can be connected to another `AsyncReset`{.firrtl} or to a `Reset`{.firrtl}.
+```firrtl
+UInt
+SInt
+Analog
+```
 
-Two enumeration types are equivalent if both have the same number of variants, and both the enumerations' i'th variants have matching names and equivalent types.
+When an inferred variant of an integer ground type is used, FIRRTL will calculate the minimum width needed for it.
+It is an error if the minimum size cannot be calculated.
 
-Two vector types are equivalent if they have the same length, and if their element types are equivalent.
+For example, the width of a multiplexer expression is the maximum of its two corresponding input widths.
 
-Two bundle types are equivalent if they have the same number of fields, and both the bundles' i'th fields have matching names and orientations, as well as equivalent types.
-Consequently, `{a:UInt, b:UInt}`{.firrtl} is not equivalent to `{b:UInt, a:UInt}`{.firrtl}, and `{a: {flip b:UInt}}`{.firrtl} is not equivalent to `{flip a: {b: UInt}}`{.firrtl}.
+The width of each primitive operation is detailed in [@sec:primitive-operations].
 
-Two property types are equivalent if they are the same concrete property type.
+### Reset Inference
+
+The uninferred `Reset`{.firrtl} type will be inferred to either a synchronous reset `UInt<1>`{.firrtl} or to an asynchronous reset `AsyncReset`{.firrtl}.
+
+The following example shows an inferred reset that will get inferred to a synchronous reset.
 
-## Width Inference
+``` firrtl
+input a : UInt<1>
+wire reset : Reset
+connect reset, a
+```
 
-For all circuit components declared with unspecified widths,
-the FIRRTL compiler will infer the minimum possible width that maintains the legality of all its incoming connections.
-If a circuit component has no incoming connections, and the width is unspecified, then an error is thrown to indicate that the width could not be inferred.
+After reset inference, `reset`{.firrtl} is inferred to the synchronous `UInt<1>`{.firrtl} type:
```

Comment:
This sentence and the previous one seem to restate the same point. I think this is actually incorrectly repeated twice.

Reply:
This was the original copy.